File: Python_Codes/sumOfNTerms.py
def sum1(A,D,N):
    if (N==0):
        return 0
    elif (N>0):
        return sum1(A,D,N-1)+A+(N-1)*D
t=int(input('No of test cases:'))
for i in range(1,t+1,1):
    print(f'\nCase-{i}:')
    A=int(input('1st element,A = '))
    D=int(input('Interval,D = '))
    N=int(input('n-terms = '))
    print('Sum of 1st n-terms:', sum1(A,D,N))

File: Prolog_Codes/sumOfNTerms.pl
findsum:- write('1st term:'), read(A), write('Interval:'), read(D),
	write('n-term:'), read(N), sum1(A, D, N, S),
	write('Sum of n terms:'), write(S).
sum1(_, _, 0, 0):-!.
sum1(A, D, N, S):-N1 is N-1, sum1(A, D, N1, S1), S is S1+A+(N-1)*D.